### DISCUSSION TITLE
Jest with Fetch and msw

### DISCUSSION THREAD

**Discussion by jkupcho** [vercel/next.js#28421] [Help] [answered]
Good morning,

I'm trying to use `msw` to mock actual endpoint calls through `jest` from within an API route and I'm receiving `fetch is not defined`. If I try actually running the application, `fetch` is available. Anyone have a quick way to make sure jest has `fetch` available for use when running tests?

Thanks!

**Comment by jkupcho** [comment, marked as answer]
I ended up giving up and using Axios on the backend. If anyone has a solution to just use `fetch` bundled with NextJS, I'm all ears.

**Reply by simonsinclair** [reply]
https://github.com/vercel/next.js/discussions/13678#discussioncomment-22383 might help you here.

We follow this advice in our `jest.setup.js`, so tests benefit from the same polyfills as our application.

**Reply by francisdaigle** [reply]
Importing [whatwg-fetch](https://www.npmjs.com/package/whatwg-fetch) in `jest.setup.js` addresses this issue as well.

**Reply by philwolstenholme** [reply]
For anyone who finds this in the future like I did, importing `whatwg-fetch` or `node-fetch` or `isomorphic-fetch` or whatever else won't give you the same version of Fetch that Next itself gives you, so your tests might run, but they won't be as close-to-real-life as they could be.
